Exception when saving game in multiplayer turn check worker

**Version**
4.1.

As reported by Vsynk#7904:

```
com.badlogic.gdx.utils.SerializationException: Error writing file: MultiplayerGames/GREECE -- gios synk dune
    at com.badlogic.gdx.utils.Json.toJson(Json.java:259)
    at com.badlogic.gdx.utils.Json.toJson(Json.java:243)
    at com.unciv.logic.GameSaver.saveGame(GameSaver.kt:113)
    at com.unciv.logic.GameSaver.saveGame(GameSaver.kt:104)
    at com.unciv.logic.GameSaver.saveGame$default(GameSaver.kt:102)
    at com.unciv.app.MultiplayerTurnCheckWorker$doWork$1.invokeSuspend(MultiplayerTurnCheckWorker.kt:291)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
    at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:279)
    at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
    at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
    at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
    at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
    at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
    at com.unciv.app.MultiplayerTurnCheckWorker.doWork(MultiplayerTurnCheckWorker.kt:256)
    at androidx.work.Worker$1.run(Worker.java:86)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String com.badlogic.gdx.Files.getLocalStoragePath()' on a null object reference
    at com.badlogic.gdx.backends.android.AndroidFileHandle.file(AndroidFileHandle.java:254)
    at com.badlogic.gdx.files.FileHandle.mkdirs(FileHandle.java:549)
    at com.badlogic.gdx.files.FileHandle.writer(FileHandle.java:353)
    at com.badlogic.gdx.utils.Json.toJson(Json.java:256)
    ... 18 more
```

--- a/android/src/com/unciv/app/MultiplayerTurnCheckWorker.kt
+++ b/android/src/com/unciv/app/MultiplayerTurnCheckWorker.kt
@@ -16,6 +16,7 @@ import androidx.core.app.NotificationCompat
 import androidx.core.app.NotificationCompat.DEFAULT_VIBRATE
 import androidx.core.app.NotificationManagerCompat
 import androidx.work.*
+import com.badlogic.gdx.Gdx
 import com.badlogic.gdx.backends.android.AndroidApplication
 import com.badlogic.gdx.backends.android.DefaultAndroidFiles
 import com.unciv.logic.GameInfo
@@ -256,6 +257,8 @@ class MultiplayerTurnCheckWorker(appContext: Context, workerParams: WorkerParame
         // We can't use Gdx.files since that is only initialized within a com.badlogic.gdx.backends.android.AndroidApplication.
         // Worker instances may be stopped & recreated by the Android WorkManager, so no AndroidApplication and thus no Gdx.files available
         val files = DefaultAndroidFiles(applicationContext.assets, ContextWrapper(applicationContext), false)
+        // GDX's AndroidFileHandle uses Gdx.files internally, so we need to set that to our new instance
+        Gdx.files = files
         gameSaver.init(files, null)
     }
 

